From 9317c9d6eba6d1d6b5eca34efff36d11cbe43d5f Mon Sep 17 00:00:00 2001
From: Brock Noland <brock@apache.org>
Date: Sat, 7 Feb 2015 00:14:56 +0000
Subject: [PATCH 0022/1164] HIVE-9411 - Improve error messages in
 TestMultiOutputFormat (Brock via Xuefu)

git-svn-id: https://svn.apache.org/repos/asf/hive/trunk@1657999 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit bca37f9d69fc48e5eaa3a76c1957bd4ad04b439a)
---
 .../hcatalog/mapreduce/TestMultiOutputFormat.java  |   14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce/TestMultiOutputFormat.java b/hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce/TestMultiOutputFormat.java
index 237ed85..042ab33 100644
--- a/hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce/TestMultiOutputFormat.java
+++ b/hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce/TestMultiOutputFormat.java
@@ -141,13 +141,19 @@ public void testMultiOutputFormatWithoutReduce() throws Throwable {
     // Verify if the configs are merged
     Path[] fileClassPaths = DistributedCache.getFileClassPaths(job.getConfiguration());
     List<Path> fileClassPathsList = Arrays.asList(fileClassPaths);
-    Assert.assertTrue(fileClassPathsList.contains(new Path(inputFile)));
-    Assert.assertTrue(fileClassPathsList.contains(new Path(dummyFile)));
+    Assert.assertTrue("Cannot find " + (new Path(inputFile)) + " in " + fileClassPathsList,
+      fileClassPathsList.contains(new Path(inputFile)));
+    Assert.assertTrue("Cannot find " + (new Path(dummyFile)) + " in " + fileClassPathsList,
+      fileClassPathsList.contains(new Path(dummyFile)));
 
     URI[] cacheFiles = DistributedCache.getCacheFiles(job.getConfiguration());
     List<URI> cacheFilesList = Arrays.asList(cacheFiles);
-    Assert.assertTrue(cacheFilesList.contains(new Path(inputFile).makeQualified(fs).toUri()));
-    Assert.assertTrue(cacheFilesList.contains(new Path(dummyFile).makeQualified(fs).toUri()));
+    URI inputFileURI = new Path(inputFile).makeQualified(fs).toUri();
+    Assert.assertTrue("Cannot find " + inputFileURI + " in " + cacheFilesList,
+      cacheFilesList.contains(inputFileURI));
+    URI dummyFileURI = new Path(dummyFile).makeQualified(fs).toUri();
+    Assert.assertTrue("Cannot find " + dummyFileURI + " in " + cacheFilesList,
+      cacheFilesList.contains(dummyFileURI));
 
     Assert.assertTrue(job.waitForCompletion(true));
 
-- 
1.7.9.5

